### PR TITLE
df-92and91 -> Support GET for the DoT valve for WELL objects

### PR DESCRIPTION
**Pull request by randypitcherii**
This PR includes:
- testing suite for `getObject` for `DoTValve`
- upgrade to `WitsmlObjects` library version 1.1.4
- support for getting 1311 and 1411 WELL objects by `UID` with field filtering
- upgrade to `DoTValve` to support sending `POST` instead of `PUT` requests to create objects

This connects to #91 
This connects to #92 

EDIT:
Chris is also adding fixes for a related issue with Drillflow responses.
This connects to #138

### CODE REVIEW

**Comment by randypitcherii** [comment]
NOTE -> this will fail travis until the 1.1.4 `WitsmlObjects` library PR is approved.

File: Makefile [hunk 1]
```diff
@@ -13,6 +13,14 @@ package:
 fastPackage:  
 	mvn -q clean package -DskipTests=true
```
Comment (randypitcherii) [inline Makefile]: cool randy makefile nonsense

File: df-server/src/main/java/com/hashmapinc/tempus/witsml/server/api/IStore.java [hunk 2]
```diff
@@ -53,7 +53,7 @@ int addToStore(
     @RequestWrapper(targetNamespace = "http://www.witsml.org/message/120")
     String getBaseMsg(@WebParam(partName= "ReturnValueIn") Short ReturnValueIn);
 
-    @WebMethod(action = "http://www.witsml.org/action/120/Store.WMLS_GetFromStore")
+    @WebMethod(action = "http://www.witsml.org/action/120/Store.WMLS_GetFromStore", operationName = "WMLS_GetFromStore")
```
Comment (randypitcherii) [inline df-server/src/main/java/com/hashmapinc/tempus/witsml/server/api/IStore.java]: fixed the soap method name

File: df-server/src/main/java/com/hashmapinc/tempus/witsml/server/api/StoreImpl.java [hunk 3]
```diff
@@ -219,26 +219,31 @@ public WMLS_GetFromStoreResponse getFromStore(
         try {
```
Comment (randypitcherii) [inline df-server/src/main/java/com/hashmapinc/tempus/witsml/server/api/StoreImpl.java]: 2 basic changes here:
- added support for the ValveUser for passing credentials to the query context
- call `valve.getObject(qc)` to get the XML string to return in the response object formatted according to the `CLIENT_VERSION` in `qc`

File: df-valve/src/main/java/com/hashmapinc/tempus/witsml/valve/dot/DotTranslator.java [hunk 4]
```diff
@@ -15,9 +15,15 @@
  */
```
Comment (randypitcherii) [inline df-valve/src/main/java/com/hashmapinc/tempus/witsml/valve/dot/DotTranslator.java]: Added method to take the query json object parsed from the soap client and merge it with the `GET` result from `DoT`. 

This sweet jamba juice relies on upgraded JSON support to `WitsmlObjects` library. This was built by yours truly in version `1.1.4`.

File: df-valve/src/main/java/com/hashmapinc/tempus/witsml/valve/dot/DotValve.java [hunk 5]
```diff
@@ -18,14 +18,18 @@
 import java.util.logging.Logger;
```
Comment (randypitcherii) [inline df-valve/src/main/java/com/hashmapinc/tempus/witsml/valve/dot/DotValve.java]: Added query logic directly in the `getObject` method with no object type checking.

I will move ALL this logic to the `Delegator` when I support `WELLBORE` querying.

I know this is dumb, but please just trust me for now. This is the result of my previous bad choices, my love for killing waffle cards, and some important PR's that are pending approval.

For more info, please just message me. But trust me, this is for the best for now.

File: df-valve/src/test/java/com/hashmapinc/tempus/witsml/valve/dot/DotValveTest.java [hunk 6]
```diff
@@ -90,6 +90,32 @@ public void createObjectWell1411() throws IOException, JAXBException{
         assertNotNull(uid);
```
Comment (randypitcherii) [inline df-valve/src/test/java/com/hashmapinc/tempus/witsml/valve/dot/DotValveTest.java]: I added a 1311 and 1411 well XML query file to test resources

I added a test for each WELL version for getting by UID. The UID is currently based on the dummy data that `Drilltest` autopopulates.

The tests simply ensure that any data is returned. More advanced testing is needed to actually check the quality of the data returned.

**Comment by cherrera2001** [comment]
Also currently the response for the query:

```xml
<wells xmlns="http://www.witsml.org/schemas/1series" version="1.4.1.1">
  <well uid="w-123">
    <name />
    <timeZone />
    <wellLocation />
  </well>
</wells>
```
Returns:
```xml
<ns0:wells xmlns:ns0="http://www.witsml.org/schemas/1series" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.witsml.org/schemas/1series http://www.witsml.org/schemas/1series/obj_well.xsd" version="1.4.1.1">
  <ns1:well xmlns:ns1="http://www.witsml.org/schemas/1series" uid="w-123">
    <ns2:name xmlns:ns2="http://www.witsml.org/schemas/1series" />
    <ns3:timeZone xmlns:ns3="http://www.witsml.org/schemas/1series" />
    <ns4:wellLocation xmlns:ns4="http://www.witsml.org/schemas/1series" />
  </ns1:well>
</ns0:wells>
```
Which has a namespace per element, even though it is the same namespace.

This should be simply the same object with the values filled in.

**Comment by cherrera2001** [comment]
Further to the above comment, the namespaces seem to only happen with 1.4.1.1, not 1.3.1.1

**Comment by cherrera2001** [comment]
So i ended up committing https://github.com/hashmapinc/Drillflow/pull/137/commits/d137a0e4f75a04fa051fe27c15d456d609ffe3df to address the merging issue, and stupidly did not unstage the changes i made for the response objects, which broke the build. I then decided rather than reverting, i would just commit the response objects in https://github.com/hashmapinc/Drillflow/pull/137/commits/5d1d1094178c68bd2314cff92d4f2425ba6729a8 to address the issues caused. So this card will now address #138 as well.

**Comment by randypitcherii** [comment]
image
Bug fix of the year!!!! Thanks, Chris!

**Pull request merged**